samples/client/petstore/ruby-faraday/lib/petstore/configuration.rb

```diff
@@ -166,8 +182,12 @@ def base_path=(base_path)
       @base_path = '' if @base_path == '/'
     end
 
-    def base_url
-      "#{scheme}://#{[host, base_path].join('/').gsub(/\/+/, '/')}".sub(/\/+\z/, '')
+    # Returns base URL for specified operation based on server settings
+    def base_url(operation = nil)
+      index = server_operation_index.fetch(operation, server_index)
+      return "#{scheme}://#{[host, base_path].join('/').gsub(/\/+/, '/')}".sub(/\/+\z/, '') if index == nil
```

Comment:
Hi @jimschubert 

By default `server_index ` is 0 here which means by default the `if index == nil` will never be true. Can we change this to a better check and set the `server_index` to zero only if the host is NULL or something? 

We have the issue where we cannot set the server because of the issue above. The caller needs to explicitly set the `server_index` to nil in order for `host` to work properly

Reply:
+1 for this issue!

Reply:
Running into this issue as well. As @laertispappas says: the `if index == nil` will never be true (well, unless people explicitly set the `operation` in the `server_operation_index` Hash with the value `nil` or set `server_index` to `nil`, both of which do not seem to make much sense. 

I think no one is using this template successfully. It seems unlikely or everyone is using per-operation servers. 

If you patch line 186 to be
```ruby
index = server_operation_index[operation]
```
then everything works fine.

Reply:
can you please file a PR with the suggested patch when you've time?

Reply:
Done: https://github.com/OpenAPITools/openapi-generator/pull/15162